roles/ocp_on_libvirt/tasks/redfish_setup.yml

```diff
@@ -20,15 +20,29 @@
     sushy_ignore_boot_device: false
     inventory_validated: true
 
-- name: Get KVM hosts UUID
-  shell: >
-    set -o pipefail; virsh list --all --name --uuid |
-    sed -e 's/^\([^ ]*\) \([^ ]*\)$/"\2": "\1",/g' |
-    tr -d '\n' |
-    sed -e 's/^\(.*\),$/{\1}/g'
-  register: all_vms
+- name: Get KVM host list
   become: true
+  register: _ool_vms
+  community.libvirt.virt:
+    command: list_vms
+
+- name: Get KVM hosts xml
+  become: true
+  register: _ool_vms_xml
+  loop: "{{ _ool_vms.list_vms }}"
+  community.libvirt.virt:
+    command: get_xml
+    name: "{{ item }}"
+
+- name: Extract UUID from xmls
+  loop: "{{ range(0, _ool_vms.list_vms | length ,1) | list }}"
+  register: _ool_host_uuid
+  community.general.xml:
+    xmlstring: "{{ _ool_vms_xml.results[item].get_xml }}"
+    xpath: /domain/uuid
+    content: text
 
 - name: Store KVM hosts UUID
-  set_fact:
-    redfish_kvm_uuid: "{{ all_vms.stdout | from_json }}"
+  loop: '{{ range(0, _ool_vms.list_vms | length ,1) | list  }}'
+  ansible.builtin.set_fact:
+    ool_redfish_kvm_uuid: "{{ redfish_kvm_uuid | default({}) | combine({ _ool_vms.list_vms[item]: _ool_host_uuid.results[item].matches[0].uuid }) }}"
```

Comment:
For some reason this task is failing now.
When testing this change, this task keeps overriding the UUIDs found.
From [this job](https://www.distributed-ci.io/jobs/807a5cf9-3571-4f61-8287-5d3845c6dd96/jobStates?sort=date&task=4e70a12c-f3ef-4442-b54d-452a6b52cbd0)

```
TASK [redhatci.ocp.ocp_on_libvirt : Store KVM hosts UUID]
0s
ok: [server06] => (item=0) => {"ansible_facts": {"ool_redfish_kvm_uuid": {"dciokd-master-2": "24a61333-b9c3-4905-9790-d0e9f70a57b2"}}, "ansible_loop_var": "item", "changed": false, "item": 0}
ok: [server06] => (item=1) => {"ansible_facts": {"ool_redfish_kvm_uuid": {"dciokd-worker-1": "18abdec5-ecde-4c57-bcfa-9c985a8067f5"}}, "ansible_loop_var": "item", "changed": false, "item": 1}
ok: [server06] => (item=2) => {"ansible_facts": {"ool_redfish_kvm_uuid": {"provisionhost": "63d6de1e-fe68-4aa2-a452-af7bbd2924ab"}}, "ansible_loop_var": "item", "changed": false, "item": 2}
ok: [server06] => (item=3) => {"ansible_facts": {"ool_redfish_kvm_uuid": {"dciokd-worker-0": "0bb9d786-19c8-445f-b555-5ef72d49831d"}}, "ansible_loop_var": "item", "changed": false, "item": 3}
ok: [server06] => (item=4) => {"ansible_facts": {"ool_redfish_kvm_uuid": {"dciokd-worker-2": "20b72422-e215-44d0-9050-e0dd3bac8073"}}, "ansible_loop_var": "item", "changed": false, "item": 4}
ok: [server06] => (item=5) => {"ansible_facts": {"ool_redfish_kvm_uuid": {"dciokd-master-1": "61c91197-c70f-4da6-9471-028c749ffb2d"}}, "ansible_loop_var": "item", "changed": false, "item": 5}
ok: [server06] => (item=6) => {"ansible_facts": {"ool_redfish_kvm_uuid": {"dciokd-master-0": "e650fffd-f559-458e-8904-ac9c447453be"}}, "ansible_loop_var": "item", "changed": false, "item": 6}
ok: [server06] => (item=7) => {"ansible_facts": {"ool_redfish_kvm_uuid": {"sno": "c7132c59-2566-44f3-86a1-d13c4a95308f"}}, "ansible_loop_var": "item", "changed": false, "item": 7}
```

You see every iteration just resets the fact.

[This is from a working job](https://www.distributed-ci.io/jobs/0a72ebac-2ceb-420a-b368-e77767013702/jobStates?sort=date&task=912849ac-faf3-4f93-83a1-71a499102b8a):

```
TASK [redhatci.ocp.ocp_on_libvirt : Get KVM hosts UUID]
1s
changed: [server07] => {"changed": true, "cmd": "set -o pipefail; virsh list --all --name --uuid | sed -e 's/^\\([^ ]*\\) \\([^ ]*\\)$/\"\\2\": \"\\1\",/g' | tr -d '\
' | sed -e 's/^\\(.*\\),$/{\\1}/g'
", "delta": "0:00:00.034175", "end": "2024-06-26 12:37:26.766958", "rc": 0, "start": "2024-06-26 12:37:26.732783", "stderr": "", "stderr_lines": [], "stdout": "{\"sno\": \"d3984a82-778f-4b2d-8adb-0f08f420682a\",\"dciokd-master-0\": \"9099a565-f894-470a-8e14-fe84ba9a3984\",\"dciokd-master-1\": \"547608b6-c5ce-49c5-af98-1fa2d9899458\",\"dciokd-master-2\": \"c65aa9c1-9598-4b83-bfce-37967200bccb\",\"dciokd-worker-0\": \"3c4741e9-afd1-4c63-8859-559d930bd15b\",\"dciokd-worker-1\": \"1156c6f3-72d9-4002-8bbe-397dc8002aba\",\"dciokd-worker-2\": \"3b29a77e-e7f9-45f0-be87-19b27ddefd2e\",\"provisionhost\": \"e121f1d1-f92a-4055-90cf-e3448634fa4b\"}", "stdout_lines": ["{\"sno\": \"d3984a82-778f-4b2d-8adb-0f08f420682a\",\"dciokd-master-0\": \"9099a565-f894-470a-8e14-fe84ba9a3984\",\"dciokd-master-1\": \"547608b6-c5ce-49c5-af98-1fa2d9899458\",\"dciokd-master-2\": \"c65aa9c1-9598-4b83-bfce-37967200bccb\",\"dciokd-worker-0\": \"3c4741e9-afd1-4c63-8859-559d930bd15b\",\"dciokd-worker-1\": \"1156c6f3-72d9-4002-8bbe-397dc8002aba\",\"dciokd-worker-2\": \"3b29a77e-e7f9-45f0-be87-19b27ddefd2e\",\"provisionhost\": \"e121f1d1-f92a-4055-90cf-e3448634fa4b\"}"]}
```
Where the UUIDs are properly encoded in the dictionary.